File: ios/Kiosk.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Kiosk, NSObject)

RCT_EXTERN_METHOD(init:(NSDictionary *)options)

RCT_EXTERN_METHOD(enable:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(disable:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Kiosk.swift
@objc(Kiosk)
class Kiosk: NSObject {
  @objc(init:)
  func `init`(options: NSDictionary) -> Void {
    
  }
  
  @objc(enable:withRejecter:)
  func enable(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(false)
  }
    
  @objc(disable:withRejecter:)
  func disable(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(false)
  }
}
